internal/shader/expr.go

```diff
@@ -281,12 +281,20 @@ func (cs *compileState) parseExpr(block *block, expr ast.Expr, markLocalVariable
 		if !ok {
 			return nil, nil, nil, false
 		}
+
+		if len(ss) != 0 {
+			stmts = append(stmts, ss...)
+		}
+
+		if len(es) == 0 {
+			return nil, nil, stmts, true
+		}
+
 		if len(es) != 1 {
 			cs.addError(e.Pos(), fmt.Sprintf("multiple-value context is not available at a callee: %s", e.Fun))
 			return nil, nil, nil, false
 		}
 		callee = es[0]
-		stmts = append(stmts, ss...)
```

Comment:
Do we really need to move this line?

Reply:
I believe so. All the changes make it work

Reply:
What happens is, that we don't have anything in `es`, but the existing code expects `len(es)` to equal 1 to avoid a multi return func, which blocks adding of our discard statement. Further, we do return something in `es` to avoid this check, then it will still error as whatever is in `es` does not have a call type.